source/robot_model/src/Model.cpp

```diff
@@ -347,56 +347,53 @@ void Model::print_qp_problem() {
+  return ((vector.array() >= lower_limits.array()).all() && (vector.array() <= upper_limits.array()).all());
+}
+
 bool Model::in_range(const state_representation::JointPositions& joint_positions) {
-  return this->in_range(joint_positions.get_positions(), this->robot_model_.lowerPositionLimit,
-                                                         this->robot_model_.upperPositionLimit);
+  return this->in_range(joint_positions.get_positions(),
+                        this->robot_model_.lowerPositionLimit,
+                        this->robot_model_.upperPositionLimit);
 }
 
 bool Model::in_range(const state_representation::JointVelocities& joint_velocities) {
-  return this->in_range(joint_velocities.get_velocities(), -this->robot_model_.velocityLimit,
-                                                            this->robot_model_.velocityLimit);
+  return this->in_range(joint_velocities.get_velocities(),
+                        -this->robot_model_.velocityLimit,
+                        this->robot_model_.velocityLimit);
 }
 
 bool Model::in_range(const state_representation::JointTorques& joint_torques) {
-  return this->in_range(joint_torques.get_torques(), -this->robot_model_.effortLimit,
-                                                      this->robot_model_.effortLimit);
+  return this->in_range(joint_torques.get_torques(),
+                        -this->robot_model_.effortLimit,
+                        this->robot_model_.effortLimit);
 }
 
 bool Model::in_range(const state_representation::JointState& joint_state) {
-  return (this->in_range(joint_state.get_positions(), this->robot_model_.lowerPositionLimit,
-                                                       this->robot_model_.upperPositionLimit)
-       && this->in_range(joint_state.get_velocities(), -this->robot_model_.velocityLimit,
-                                                         this->robot_model_.velocityLimit)
-       && this->in_range(joint_state.get_torques(), -this->robot_model_.effortLimit,
-                                                       this->robot_model_.effortLimit));
+  return (this->in_range(static_cast<state_representation::JointPositions>(joint_state))
+      && this->in_range(static_cast<state_representation::JointVelocities>(joint_state))
+      && this->in_range(static_cast<state_representation::JointTorques>(joint_state)));
 }
 
-bool Model::in_range(const Eigen::VectorXd& vector, const Eigen::VectorXd& lower_limits,
-                                                    const Eigen::VectorXd& upper_limits) {
-
-  return ((vector.array() >= lower_limits.array()).all() && (vector.array() <= upper_limits.array()).all());
+Eigen::VectorXd Model::clamp_in_range(const Eigen::VectorXd& vector,
+                                      const Eigen::VectorXd& lower_limits,
+                                      const Eigen::VectorXd& upper_limits) {
+  return lower_limits.cwiseMin(upper_limits.cwiseMax(vector));
```

Comment:
The fact that the tests pass makes me doubt myself a lot but I'm pretty sure that this is incorrect. The `cwiseMin` and `cwiseMax` should be the other way round. Otherwise we always end up with the lower limits (which is of course in range)
```
Eigen::Vector2d upper(1,2);
Eigen::Vector2d lower(0, 0);
Eigen::Vector2d tester(1.1, 1.9);
std::cout << lower.cwiseMax(upper.cwiseMin(tester)).transpose() << std::endl;
std::cout << lower.cwiseMin(upper.cwiseMax(tester)).transpose() << std::endl;
```
output:
```
1 1.9
0 0
```

Reply:
You are right it should be like this. I did it fast and I did not check it well.
In the test I looked if it is in the range or not after clamping, for this reason it passes the test

Reply:
@patr3 can you correct this? This way yous still appears in contributors of this

Reply:
done